pkg/cfn/manager/api.go

```diff
@@ -225,8 +226,25 @@ func (c *StackCollection) DeleteStack(name string) (*Stack, error) {
 		fmt.Sprintf("%s:%s", ClusterNameTag, c.spec.Metadata.Name))
 }
 
-// WaitDeleteStack kills a stack by name and waits for DELETED status
-func (c *StackCollection) WaitDeleteStack(name string) error {
+// WaitDeleteStack kills a stack by name and waits for DELETED status;
+// any errors will be written to errs channel, when nil is written,
+// assume completion, do not expect more then one error value on the
+// channel, it's closed immediately after it is written to
+func (c *StackCollection) WaitDeleteStack(name string, errs chan error) error {
+	i, err := c.DeleteStack(name)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("waiting for stack %q to get deleted", *i.StackName)
+
+	go c.waitUntilStackIsDeleted(i, errs)
+
+	return nil
+}
+
+// BlockingWaitDeleteStack kills a stack by name and waits for DELETED status
+func (c *StackCollection) BlockingWaitDeleteStack(name string) error {
```

Comment:
We add `Blocking` to the name of the old function, because that's what it is actually. We keep using it where it was used previously, but use new non-blocking version above when we need to delete all nodegroups in parallel and run a task for each of the nodegroups.